pkg/nvidia/Dockerfile

```
@@ -0,0 +1,102 @@
+# syntax=docker/dockerfile-upstream:1.5.0-rc2-labs
+
+# Copyright (c) 2024 Zededa, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+FROM lfedge/eve-alpine:82df60e43ab9f8c935584b8c7b4d0a4b0271d608 as build-base
+ENV BUILD_PKGS="autoconf automake build-base coreutils gettext git glib-dev go libtool libmd-dev ncurses-dev tar xz-dev yq zstd-dev"
+
+RUN eve-alpine-deploy.sh
+
+# This is a arm64 (only) package, nothing to do for other architectures
+FROM build-base as build-amd64
+FROM build-base as build-riscv64
+
+FROM build-base as build-arm64
+
+# Setup git to apply patches
+RUN git config --global user.email 'builder@projecteve.dev' && \
+    git config --global user.name 'Project EVE'
+
+# Jetson Linux tarball
+ENV JETSON_LINUX=https://developer.nvidia.com/downloads/embedded/l4t/r35_release_v5.0/release/jetson_linux_r35.5.0_aarch64.tbz2
+
+ADD ${JETSON_LINUX} /jetson_linux.tbz2
+
+# We need to build dpkg in order to get zstd compression support
+ENV DPKG_REVISION=1.22.6
+ADD --keep-git-dir=true https://salsa.debian.org/dpkg-team/dpkg.git#${DPKG_REVISION} /dpkg
+WORKDIR /dpkg
+RUN ./autogen && \
+    ./configure --prefix=/ && \
+    make -j "$(nproc)" && \
+    make install
+
+# CDI Files
+COPY cdi /cdi
+
+# Extract tarball
+WORKDIR /
+RUN tar -xjf /jetson_linux.tbz2
+
+# Build nvidia-ctk
+ENV NVIDIA_CONTAINER_TOOLKIT_REV=v1.16.0
+ADD --keep-git-dir=true https://github.com/NVIDIA/nvidia-container-toolkit.git#${NVIDIA_CONTAINER_TOOLKIT_REV} /nvct
+
+WORKDIR /nvct
+
+COPY patches/* /nvct
+RUN for x in /nvct/*.patch; do \
+        git am "$x" ; \
+    done
+
+RUN mkdir -p dist && \
+    make cmd-nvidia-ctk && cp nvidia-ctk dist/ && \
+    make cmd-nvidia-cdi-hook && cp nvidia-cdi-hook dist/
```

```suggestion
RUN mkdir -p dist
RUN make cmd-nvidia-ctk && cp nvidia-ctk dist/
RUN make cmd-nvidia-cdi-hook && cp nvidia-cdi-hook dist/
```
not everything has to be in one `RUN` - this way it is more readable (subjectively) and it can cache more (objectively)

---

yeah, I agree, in this case it's better to break into multiple RUNs... it's done.